### DISCUSSION TITLE
Help Request: Reset API to show data from server while using useFieldaarray

### DISCUSSION THREAD

**Discussion by sunilbittu** [react-hook-form/react-hook-form#2592] [Q&A]
hi can someone help me to set data from an api to useFieldArray items, after reading the previous discussions i came to know that we need to use resetApi and useeffect to set the data to defaultValues.
Please tell me what's the mistake am doing in below sandbox.
https://codesandbox.io/s/tender-brattain-9y1cd?file=/src/ArrayField.js

thank you,

**Comment by bluebill1049** [comment]
<img width="851" alt="Screen Shot 2020-08-14 at 8 23 55 am" src="https://user-images.githubusercontent.com/10513364/90193053-807c7980-de07-11ea-8ef5-bc8c692af615.png">
looks like data is empty :(

**Comment by bluebill1049** [comment]
also for conditional render inside field array, you need to toggle the css visibility.

**Comment by bluebill1049** [comment]
<img width="1073" alt="Screen Shot 2020-08-14 at 8 26 42 am" src="https://user-images.githubusercontent.com/10513364/90193240-e36e1080-de07-11ea-9979-af314040b1d6.png">

looks like there are more problems at your components as well :)

**Reply by sunilbittu** [reply]
hi sir i have updated the sandbox
https://codesandbox.io/s/tender-brattain-9y1cd?file=/src/ArrayField.js
please help me

**Reply by bluebill1049** [reply]
Take a look this example: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-1scf3?file=/src/index.js
